test/shmemx/put_signal.c

```diff
@@ -86,6 +86,41 @@ int main(int argc, char *argv[])
             errors++;
         }
     } 
+
+    shmem_barrier_all();
+
+    for (i = 0; i < MSG_SZ; i++)
+        target[i] = 0;
+
+    sig_addr = 0;
+    int local_counter = 0;
+
+    shmem_barrier_all();
+    if (me == 0) {
+        shmemx_long_put_signal(target, source, MSG_SZ, &sig_addr, 1, SHMEM_SIGNAL_ADD, dest);
+        local_counter++;
+#if defined(__STDC_VERSION__) && __STDC_VERSION__ >= 201112L
```

Comment:
All these checks for C11 seem a little strange, especially since we don't do it on `shmemx_long_put_signal`.  Can't we do something like this at the top of the tests?

```c
#if defined(__STDC_VERSION__) && __STDC_VERSION__ >= 201112L
#define SHMEM_WAIT_UNTIL  shmem_wait_until
#define SHMEM_PUT_SIGNAL  shmemx_put_signal
#else
#define SHMEM_WAIT_UNTIL shmem_uint64_wait_until
#define SHMEM_PUT_SIGNAL shmemx_long_put_signal
#endif
```

Side note: when I try this with Clang, I get a nasty error from the generic selection.  However, when I remove [this `default: shmem_ctx_c11_generic_selection_failed` line here](https://github.com/Sandia-OpenSHMEM/SOS/blob/master/mpp/shmemx.h4#L483), it works fine...  Note that [`shmem_put` and (some) friends](https://github.com/Sandia-OpenSHMEM/SOS/blob/master/mpp/shmem.h4#L357) do not have this `default:` case either...  I feel like @jdinan and I have seen this before, but I don't remember the reason for this.

I suppose another more pertinent comment I'm making is that the C11 `shmem_put_signal` interfaces don't seem to be covered.

Reply:
They are currently defined as `shmemx_put_signal`. Are you referring to something else?

Reply:
Ok. I spoke too soon, I see the difference, But, are you saying you are seeing these errors only for put-signal, not for NBI atomics? They should have the same error too, right?

Reply:
You're right @wrrobin, one might expect the NBI atomics to show a similar error, but I just tried it out and don't see any issue, strangely.

Anyway, I think we can ignore my suggestion above and I'll file an issue for this.